Adds `get_contours` and supports "method" key word a keyword argument to select between `contourpy` and `scikit-image` (#7760)
sunpy/sunpy: changelog/7760.deprecation.rst

> The :meth:`sunpy.map.GenericMap.contour` method, which still uses skimage, will now raise a deprecation warning. Please use :meth:`sunpy.map.GenericMap.get_contours`, which defaults to using ContourPy, instead.

Reviewer: ```suggestion
:meth:`sunpy.map.GenericMap.contour` method has now been deprecated in favor of :meth:`sunpy.map.GenericMap.get_contours`.
```
